django/contrib/admin/sites.py

```diff
@@ -65,10 +65,14 @@ def __init__(self, name='admin'):
         self._global_actions = self._actions.copy()
         all_sites.add(self)
 
-    def check(self):
+    def check(self, app_configs):
+        if app_configs is not None:
+            app_configs = set(app_configs)  # Speed up lookups below
+
         errors = []
         for admin_obj in six.itervalues(self._registry):
-            errors.extend(admin_obj.check())
+            if app_configs is None or admin_obj.model._meta.app_config in app_configs:
```

Comment:
I think the `app_configs` need to be initialized if they're `None` so that overridding `INSTALLED_APPS` has a limiting effect here. E.g.,this might work:
```python
if app_configs is None:
    apps.get_app_configs()
```